File: app/src/main/java/com/theappsolutions/boilerplate/other/exceptions/ApiException.java
package com.theappsolutions.boilerplate.other.exceptions;

/**
 * @author Severyn Parkhomenko s.parkhomenko@theappsolutions.com
 * @copyright (c) 2018 TheAppSolutions. (https://theappsolutions.com)
 */
public class ApiException extends UserReadableException {

    private int code;

    public ApiException(int code, String message) {
        super(message);
        this.code = code;
    }

    public ApiException(int code, String message, Throwable cause) {
        super(message, cause);
        this.code = code;
    }

    public ApiException(int code, Throwable cause) {
        super(cause);
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}

File: app/src/main/java/com/theappsolutions/boilerplate/other/exceptions/ApiException.kt
package com.theappsolutions.boilerplate.other.exceptions

/**
 * @author Severyn Parkhomenko s.parkhomenko@theappsolutions.com
 * @copyright (c) 2018 TheAppSolutions. (https://theappsolutions.com)
 */
class ApiException : UserReadableException {

    var code: Int = 0
        private set

    constructor(code: Int, message: String)
            : super(message) {
        this.code = code
    }

    constructor(code: Int, message: String, cause: Throwable)
            : super(message, cause) {
        this.code = code
    }

    constructor(code: Int, cause: Throwable)
            : super(cause) {
        this.code = code
    }
}
